### DISCUSSION TITLE
input string length too short for URLs

### DISCUSSION THREAD

**Discussion by mhiggins-SA** [OpenAPITools/openapi-diff#407] [General]
the 2 spec input strings accept URL's - but the overall string length seems to truncate long (last) URLs.

**Comment by joschi** [comment]
@mhiggins-SA How long of URLs are we talking about?

Could you please provide a reproducible example of this?

**Comment by mhiggins-SA** [comment]
docker run --rm -t -v localdrive:/localdrive                                                        \
openapitools/openapi-diff:latest --html /localdrive/api-diff.html                                   \
https://api.swaggerhub.com/apis/mhiggins-sa/Petstore/1.0.0/swagger.yaml?resolved=true&flatten=false \
https://api.swaggerhub.com/apis/mhiggins-sa/Petstore/2.0.0/swagger.yaml?resolved=true&flatten=false



---
Micheál Higgins   |   Senior Solutions Architect
m: +353.86.235.4599
e: ***@***.******@***.***>
***@***.***<https://smartbear.com/>

smartbear.com<https://www.smartbear.com/>
***@***.***<https://www.facebook.com/smartbear>  ***@***.*** <https://www.linkedin.com/company/smartbear>   ***@***.*** <https://twitter.com/SmartBear>   ***@***.*** <https://www.youtube.com/channel/UC3iDZqrLWQZ4dh8zn2rU9hA>

**Reply by joschi** [reply]
The issue here is that the ampersand (`&`) character is interpreted by your shell. 😉
See also: https://bashitout.com/2013/05/18/Ampersands-on-the-command-line.html 

Try quoting the URLs:
```text
# docker run --rm -t openapitools/openapi-diff:latest "https://api.swaggerhub.com/apis/mhiggins-sa/Petstore/1.0.0/swagger.yaml?resolved=true&flatten=false" "https://api.swaggerh
ub.com/apis/mhiggins-sa/Petstore/2.0.0/swagger.yaml?resolved=true&flatten=false"
==========================================================================
==                            API CHANGE LOG                            ==
==========================================================================
                             Swagger Petstore                             
--------------------------------------------------------------------------
--                              What's New                              --
--------------------------------------------------------------------------
- GET    /mikeie/rocks

--------------------------------------------------------------------------
--                            What's Changed                            --
--------------------------------------------------------------------------
- PUT    /pet
  Request:
        - Changed application/json
          Schema: Broken compatibility
        - Changed application/xml
          Schema: Broken compatibility
- POST   /pet
  Request:
        - Changed application/json
          Schema: Broken compatibility
        - Changed application/xml
          Schema: Broken compatibility
- GET    /pet/findByStatus
  Return Type:
    - Changed 200 OK
      Media types:
        - Changed application/json
          Schema: Backward compatible
        - Changed application/xml
          Schema: Backward compatible
- GET    /pet/findByTags
  Return Type:
    - Changed 200 OK
      Media types:
        - Changed application/json
          Schema: Backward compatible
        - Changed application/xml
          Schema: Backward compatible
- GET    /pet/{petId}
  Return Type:
    - Changed 200 OK
      Media types:
        - Changed application/json
          Schema: Backward compatible
        - Changed application/xml
          Schema: Backward compatible
--------------------------------------------------------------------------
--                                Result                                --
--------------------------------------------------------------------------
                 API changes broke backward compatibility                 
--------------------------------------------------------------------------
```

**Reply by mhiggins-SA** [reply]
If I remove the flatten query parameter from both URLs it works. But the limit will cause issues when there are longer API names, organization names and host FQDNs.

**Reply by joschi** [reply]
Please quote the URLs properly in your shell. You won't have to remove any query parameters.

**Comment by mhiggins-SA** [comment]
I’ll try url encoding the strings. Not sure I want to use quotes given a pipeline is building the docker run dynamically.

Thanks for catching this !!!

--
Micheál Higgins
Senior Solutions Architect
SmartBear Software

Sent from my iPad



On 8 Sep 2022, at 20:59, Jochen Schalanda ***@***.***> wrote:

﻿

The issue here is that the ampersand (&) character is interpreted by your shell. 😉
See also: https://bashitout.com/2013/05/18/Ampersands-on-the-command-line.html<https://bashitout.com/2013/05/18/Ampersands-on-the-command-line.html>

Try quoting the URLs:

# docker run --rm -t openapitools/openapi-diff:latest "https://api.swaggerhub.com/apis/mhiggins-sa/Petstore/1.0.0/swagger.yaml?resolved=true&flatten=false<https://api.swaggerhub.com/apis/mhiggins-sa/Petstore/1.0.0/swagger.yaml?resolved=true&flatten=false>" "https://api.swaggerh
ub.com/apis/mhiggins-sa/Petstore/2.0.0/swagger.yaml?resolved=true&flatten=false<http://ub.com/apis/mhiggins-sa/Petstore/2.0.0/swagger.yaml?resolved=true&flatten=false>"
==========================================================================
==                            API CHANGE LOG                            ==
==========================================================================
                             Swagger Petstore
--------------------------------------------------------------------------
--                              What's New                              --
--------------------------------------------------------------------------
- GET    /mikeie/rocks

--------------------------------------------------------------------------
--                            What's Changed                            --
--------------------------------------------------------------------------
- PUT    /pet
  Request:
        - Changed application/json
          Schema: Broken compatibility
        - Changed application/xml
          Schema: Broken compatibility
- POST   /pet
  Request:
        - Changed application/json
          Schema: Broken compatibility
        - Changed application/xml
          Schema: Broken compatibility
- GET    /pet/findByStatus
  Return Type:
    - Changed 200 OK
      Media types:
        - Changed application/json
          Schema: Backward compatible
        - Changed application/xml
          Schema: Backward compatible
- GET    /pet/findByTags
  Return Type:
    - Changed 200 OK
      Media types:
        - Changed application/json
          Schema: Backward compatible
        - Changed application/xml
          Schema: Backward compatible
- GET    /pet/{petId}
  Return Type:
    - Changed 200 OK
      Media types:
        - Changed application/json
          Schema: Backward compatible
        - Changed application/xml
          Schema: Backward compatible
--------------------------------------------------------------------------
--                                Result                                --
--------------------------------------------------------------------------
                 API changes broke backward compatibility
--------------------------------------------------------------------------


—
Reply to this email directly, view it on GitHub<https://github.com/OpenAPITools/openapi-diff/discussions/407#discussioncomment-3599841>, or unsubscribe<https://github.com/notifications/unsubscribe-auth/ANMNL5XIJHGPNE3UUW4DAXDV5JATFANCNFSM6AAAAAAQICNRJU>.
You are receiving this because you were mentioned.Message ID: ***@***.***>

We hope you found this email of interest; however, click here if you wish to unsubscribe<http://www2.smartbear.com/SubscriptionCenter.html?utm_source=outlook&utm_medium=email&utm_content=emailsig> or manage your email preferences. Privacy Policy.<https://smartbear.com/privacy/>

Disclaimer

The information contained in this communication from the sender is confidential. It is intended solely for use by the recipient and others authorized to receive it. If you are not the recipient, you are hereby notified that any disclosure, copying, distribution or taking action in relation of the contents of this information is strictly prohibited and may be unlawful.
